osu.Game/Skinning/Editor/SkinEditor.cs

```diff
@@ -330,6 +352,68 @@ private void revert()
                 // add back default components
                 getTarget(t.Target).Reload();
             }
+
+            RemoveMenuBackground();
+        }
+
+        private void removeMenuBackground()
+        {
+            RemoveMenuBackground();
+
+            onScreenDisplay?.Display(new SkinEditorToast(ToastStrings.MenuBackgroundChanged, CommonStrings.Default));
+        }
+
+        public void SetMenuBackground(FileInfo fileInfo)
+        {
+            using MemoryStream menuBackgroundStream = new MemoryStream();
+            using (var stream = fileInfo.OpenRead())
+                stream.CopyTo(menuBackgroundStream);
+            menuBackgroundStream.Seek(0, SeekOrigin.Begin);
+
+            Texture previewTexture = Texture.FromStream((skins as IO.IStorageResourceProvider).Renderer, menuBackgroundStream);
+            menuBackgroundStream.Seek(0, SeekOrigin.Begin);
+
+            currentSkin.Value.SkinInfo.PerformWrite(skinInfo =>
+            {
+                using (menuBackgroundStream)
+                    skins.AddFile(skinInfo, menuBackgroundStream, $"menu-background{fileInfo.Extension}");
+            });
+            realm.Run(r => r.Refresh());
+
+            if (previewBackground != null)
+                Remove(previewBackground, true);
+
+            onScreenDisplay?.Display(new SkinEditorToast(ToastStrings.MenuBackgroundChanged, fileInfo.Name));
+            game?.PerformFromScreen(screen =>
+            {
+                (screen as MainMenu)?.ApplyToBackground(b =>
+                {
+                    if (b is BackgroundScreenDefault target)
+                        AddInternal(previewBackground = new MenuBackgroundPreview(target, previewTexture));
```

Comment:
I don't understand the point of this "background preview" thing. The skin is being changed instantaneously, so why bother with adding so much complexity? Not only you have to add and remove this "preview", you also have to "lock" and "unlock" the background. This seems too complicated for what it really needs to be.

In my ideal implementation of this:

* `MenuBackgroundPreview` does not exist
* `BackgroundScreenDefault.{Lock,Unlock}()` do not exist
* `BackgroundScreenDefault.GetBackgroundFallbackTextureName()` is private again

Unless a very compelling reason is provided for structuring things this way I can't find myself agreeing with any of this.